app/models/zoom_meeting.rb

```diff
@@ -74,7 +75,14 @@ def create_participant_objects
   def uniq_participants_best_time(participants)
     grouped_by_name = participants.group_by(&:name)
     participants_best_time = grouped_by_name.map do |name, participant_records|
-      participant_records.min_by(&:join_time)
+      earliest_join_time = participant_records.min_by(&:join_time)
+      earliest_join_time.duration = calculate_duration(participant_records)
+      earliest_join_time
     end
   end
+
+  def calculate_duration(participant_records) 
+    meeting_duration = self.duration * 60
+    ((participant_records.sum(&:duration).to_f/meeting_duration) * 100 ).round
+  end 
```

Comment:
We will get more accurate data if we store the raw number of minutes the student was in the class, and then when we display that we can make this calculation. I'm thinking this should live in application helper as a view helper and be called in the view.